client/src/templates/Challenges/utils/frame.js

```diff
@@ -103,8 +103,11 @@ const initTestFrame = frameReady => ctx => {
     // default for classic challenges
     // should not be used for modern
     const code = sources && 'index' in sources ? sources['index'] : '';
-    // provide the file name and get the original source
-    const getUserInput = fileName => toString(sources[fileName]);
+    const getUserInput = fileName => {
+      return fileName === 'url'
```

Comment:
IMO, this is not the right place for such things. It may be better to do this while storing these values ​​in the store.

Reply:
In principle, yes, but isn't simple as `redux-form` updates the store as the user types.  If you remove slashes before storing a url, the user can never type something like example.com/about because they get stuck at example.com.

I can see two options.  Normalize in the selector or normalize in `handleSubmit`.

Doing it via the selector is trivial, but does mean the store will have trailing slashes.

Doing it in `handleSubmit` is a bit messy. since you'd need to do something similar to `ProjectForm`.  Add an action like `updateBackendForm` and use that to change the form's store before executing the challenge.

Do you think it's worth adding the extra complexity or would normalizing the selector be okay?

Reply:
> Doing it in handleSubmit is a bit messy. since you'd need to do something similar to ProjectForm. Add an action like updateBackendForm and use that to change the form's store before executing the challenge.

I think this solution is ok. But without "change the form's store before executing the challenge". I propose to add to the `store` a `backendFormValues`, update them onSubmit and get these values in `backendFormValuesSelector`.

Reply:
That seems to be working out okay.  My latest commit reverts the original changes and does everything via `submit`.